scipio/src/sys/posix_buffers.rs

```diff
@@ -8,11 +8,13 @@
 // but in the near future they can be coming from memory-areas
 // that are pre-registered for I/O uring.
 
+use std::ptr;
+
 use aligned_alloc::{aligned_alloc, aligned_free};
 
 #[derive(Debug)]
 pub struct PosixDmaBuffer {
-    data: *mut u8,
+    data: ptr::NonNull<u8>,
```

Comment:
I'll adding this one to my toolbox. I've been adding way too many raw pointers all over =(